prow/config/jobs.go

```diff
@@ -231,6 +231,9 @@ type Periodic struct {
 	Cron string `json:"cron,omitempty"`
 	// Tags for config entries
 	Tags []string `json:"tags,omitempty"`
+	// IntervalWaitsForCompletion waits for current job to finish before for interval calculation starts
```

Comment:
This relies on the `interval` field that is supposed to be deprecated 5 years ago as of #5284. @krzyzacy do you still recall the reason for deprecation?

Reply:
My understanding was that direct access to `Interval` was deprecated in favor of getter and setter. That being said, from what I can tell, `interval` is still being used pretty much everywhere (at least in OpenShift).

Reply:
@chaodaiG OpenShift is still using `interval` heavily (even though we also have a lot of uses of `cron`), and I don't believe we even noticed it had been deprecated. Perhaps these options can continue to co-exist?

Reply:
I'd personally be very annoyed if `interval` goes away. I've tried to explain in the [related issue](https://github.com/kubernetes/test-infra/issues/26478) why I think `cron` is not a replacement for `interval`. Basically what you gain in flexibility with `cron`, you lose it in ease of maintenance. It can also be a lot heavier on the CI infrastructure if you need to re-arrange a bunch of jobs, which is exactly what I'm trying to avoid with this PR.

@krzyzacy, if you have strong opinion on the deprecation of `Interval`, or this patch in particular, please let us hear it.